wire/msgtx.go

```diff
@@ -600,34 +780,44 @@ func (msg *MsgTx) decodePrefix(r io.Reader, pver uint32) error {
 	}
 
 	// TxOuts.
+	var totalScriptSize uint64
 	txOuts := make([]TxOut, count)
 	msg.TxOut = make([]*TxOut, count)
 	for i := uint64(0); i < count; i++ {
+		// The pointer is set now in case a script buffer is borrowed
+		// and needs to be returned to the pool on error.
 		to := &txOuts[i]
+		msg.TxOut[i] = to
 		err = readTxOut(r, pver, msg.Version, to)
 		if err != nil {
+			returnMemory()
 			return err
 		}
-		msg.TxOut[i] = to
+		totalScriptSize += uint64(len(to.PkScript))
 	}
 
 	// Locktime and expiry.
 	msg.LockTime, err = binarySerializer.Uint32(r, littleEndian)
 	if err != nil {
+		returnMemory()
 		return err
 	}
 
 	msg.Expiry, err = binarySerializer.Uint32(r, littleEndian)
 	if err != nil {
+		returnMemory()
 		return err
 	}
 
+	writeTxOutScriptsToMsgTx(msg, totalScriptSize)
```

Comment:
I think this needs a comment to explain what's going on since this would be quite unintuitive for a reader that is not already familiar with this code.